trust-quorum/gfss/src/gf256.rs

```diff
@@ -0,0 +1,302 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Implementation of Finite Field GF(2^8) aka GF(256)
+//!
+//! We use the Rijndael (AES) polynomial (x^8 + x^4 + x^3 + x + 1) as our
+//! irreducible polynomial.
+//!
+//! We only implement enough operations so that we can implement
+//! shamir secret sharing. This keeps the surface area small and
+//! minimizes the amount of auditing we need to do.
+//!
+//! For a basic overview of galois fields, the docs from the
+//! [gf256 crate](https://docs.rs/gf256/0.3.0/gf256/gf/index.html)
+//! are excellent.
+//!
+//! For a more comprehensive introduction to the math, these
+//! [lecture notes](https://web.stanford.edu/~marykw/classes/CS250_W19/readings/Forney_Introduction_to_Finite_Fields.pdf)
+//! are handy.
+
+// Don't tell me what operations to use in my implementations
+#![allow(clippy::suspicious_arithmetic_impl)]
+
+use core::fmt::{self, Binary, Display, Formatter, LowerHex, UpperHex};
+use core::ops::{Add, AddAssign, Div, Mul, MulAssign, Sub};
```

Comment:
Is this meant to be no-std?

Reply:
There's nothing really in this file that can't be `#[no_std]`, but the rest of the crate isn't. That was kinda ambitious on my part. I still think it's fine to use `core::` instead of `std::` here, but happy to change it if there's a reason.